Gems/Atom/RPI/Code/Include/Atom/RPI.Edit/Shader/ShaderVariantListSourceData.h

```diff
@@ -77,6 +78,9 @@ namespace AZ
             AZStd::string m_shaderFilePath; // .shader file.
             AZStd::vector<VariantInfo> m_shaderVariants;
 
+            //! non-mandatory list of option-names that have been gatherd from material properties.
+            //! This allows for UX hints in differenciating system options from material options.
+            AZStd::unordered_set<Name> m_materialOptionsHint;
```

Comment:
That method is subject to evolve in coming works, because for the "enumeration" script as you said, we'll need to mask out the material options, so this will need to be more robust, it will probably come from C++ in the following works.
I'd suggest to keep this like this for now.